src/Evolution/Systems/CurvedScalarWave/Worldtube/PunctureField.hpp

```diff
@@ -0,0 +1,60 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace CurvedScalarWave::Worldtube {
+/*!
+ * \brief Computes the puncture/singular field \f$\Psi^\mathcal{P}\f$ of a
+ * scalar charge in circular orbit around a Schwarzschild black hole as
+ * described in \cite Detweiler2003.
+ *
+ * \details The field is computed using a Detweiler-Whiting singular
+ * Green's function and perturbatively expanded in the geodesic distance from
+ * the particle. It solves the inhomogeneous wave equation
+ *
+ * \f{align*}{
+ * \Box \Psi^\mathcal{P} = -4 \pi q \int \sqrt{-g} \delta^4(x^i, z(\tau)) d \tau
+ * \f}
+ *
+ * where \f$q\f$ is the scalar charge and \f$z(\tau)\f$ is the worldline of the
+ * particle. The expression is expanded up to a certain order in geodesic
+ * distance and transformed to Kerr-Schild coordinates.
+ *
+ * The function given here assumes that the particle has scalar charge \f$q=1\f$
+ * and is on a fixed geodesic orbit  with
+ * \f$z(t) = (R \cos{\omega t}, R \sin{\omega t}, 0)\f$ around a Schwarzschild
+ * black hole, where \f$R\f$ is the orbital radius and
+ * \f$ \omega = R^{-3/2} \f$. It returns the singular field at the requested
+ * coordinates as well as its time and spatial derivative.
+ *
+ * \note The expressions were computed with Mathematica and optimized by
```

Comment:
Are there plans to make the notebook(s) public as part of the paper? We can add them into the spectre repo, not necessarily in this PR. Probably somewhere in `docs` or in the same directory as this file is :shrug:

Reply:
Mathematica notebooks are kind of awful to have in a code repo.  We used to have some in SpEC and removed them because (quoting the commit message) "These notebook files are huge and interfere with git grep because they are quasi-binary."  To be more specific, several MB with much of that base64 encoded data.

Reply:
Will the notebook get published with the paper? Then we can reference the supplementary files of the paper here.

Reply:
Let me discuss that with the Southhampton folks. Alternatively I can just put it in a separate public github repo and link to it here. Burden of maintenance would fall on me but it seems unlikely to change.

Reply:
Sounds good, let's not block this PR on making any notebooks public.